New Rule: Links with identical names and context serve equivalent purpose (fd3a94) (#693)
act-rules/act-rules.github.io: _rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

accessibility_requirements:
  wcag20:2.4.4: # Link Purpose (In Context) (A)
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS Styling
authors:
  - Anne Thyme Nørregaard
---

## Applicability

This rule applies to any set of any two or more HTML or SVG elements that have the [semantic role](#semantic-role) of `link`, or a role that inherits from the `link` role, are [included in the accessibility tree](#included-in-the-accessibility-tree), and that have [matching](#matching-characters) [accessible names](#accessible-name) that do not only consist of [whitespace](#whitespace), and have [matching](#matching-characters) [programmatically determined link contexts](#programmatically-determined-link-context).

**Note:** The test target for this rule is the full set of link elements that share the same [matching](#matching-characters) [accessible name](#accessible-name) and [matching](#matching-characters) [programmatically determined link contexts](#programmatically-determined-link-context).

> ## Expectation

Reviewer: According to SC 2.4.4 the purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context.

I do not think the test rule tests this. According to the test rule, links with matching names should resolve to the same or equivalent resource. If the linked resource(s) have a different purpose than what is described in the link names, this would pass the test rule, but fail SC 2.4.4.

According to the SC 2.4.4 understanding article "it is a best practice for links with the same destination to have consistent text (and this is a requirement per Success Criterion 3.2.4 for pages in a set). It is also a best practice for links with different purposes and destinations to have different link text."

In my opinion, this test rule is more relevant for SC 3.2.4 Consistent Identification. Either way, SC 3.2.4 requires the identification to be consistent, but not necessarily identical.

Reviewer 2: But this rule also allows for "equivalent resources" not just "same resource", so it goes beyond the best practice and is actually a good basis to fail the SC on if it doesn't link to the same or equivalent resource despite having the same accessible name and context right?

```html
<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html">Contact us</a>
<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page3.html">Contact us</a>
```

#### Passed Example 8

Links created via scripting with explicit role of link, but lead to the same resource:

```html
<span
	role="link"
	tabindex="0"
	onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html'"
>
	Link text
</span>

<span
	role="link"
	tabindex="0"
	onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html'"
>
	Link text
</span>
```

#### Passed Example 9

A set of two SVG `<a>` elements have the same accessible name and link to the same resource.

```html
<svg
	viewBox="0 0 100 100"
	xmlns="http://www.w3.org/2000/svg"
	xmlns:xlink="http://www.w3.org/1999/xlink"
>
	<a href="http://facebook.com" aria-label="Follow us">
		<circle cx="50" cy="40" r="35" />
	</a>

	<a href="http://facebook.com">
		<text x="50" y="90" text-anchor="middle">
			Follow us
		</text>
	</a>
</svg>
```

### Failed

#### Failed Example 1

> 

Reviewer 3: aren't all the listed failed examples exempt since those links would be confusing/ambiguous to *all* users? e.g. how would a sighted mouse user not be confused by them versus an AT user?

Reviewer 4: That rule is Work in Progress, so anything in it might need to change ;-)
(and locally, it's waiting for #555 to be finished so that we can reuse the definition of context for the links…)

---
id: fd3a94
name: Links with identical accessible names and context serve equivalent purpose
rule_type: atomic
description: |
  This rule checks that links with identical accessible names and context resolve to the same resource or equivalent resources.
accessibility_requirements:
  wcag20:2.4.4: # Link Purpose (In Context) (A)
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS Styling
  - Language
acknowledgements:
  authors:
    - Anne Thyme Nørregaard
    - Carlos Duarte
---

## Applicability

This rule applies to any set of any two or more HTML or SVG elements which

- have the [semantic role][] of `link`, or a role that inherits from the `link` role; and
- are in the same [web page (HTML)][]; and
- are [included in an accessibility tree][included in the accessibility tree]; and

> - that have [matching][] [accessible names][accessible name] that are not empty (`""`); and

Reviewer 4: ```suggestion
- have [matching][] [accessible names][accessible name] that are not empty (`""`); and
```

> - that have the same [programmatically determined link context][].

Reviewer 4: ```suggestion
- have the same [programmatically determined link context][].
```

	</p>
</html>
```

#### Passed Example 6

These two links go to pages that contain different amounts of information and/or differently worded information, but fulfill the same purpose in relation to the link.

```html
<html lang="en">
	<p>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html">Call us</a>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page2.html">Call us</a>
	</p>
</html>
```

#### Passed Example 7

These two links go to pages that have the same advertised key content but use different layouts.

```html
<html lang="en">
	<p>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html">Contact us</a>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page3.html">Contact us</a>
	</p>
</html>
```

#### Passed Example 8

These two HTML `span` element have an [explicit role][] of link, and lead to the [same resource][].

```html
<html lang="en">
	<p>
		<span
			role="link"
			tabindex="0"
			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html'"
		>

> 			Link text

Reviewer 4: We should use a descriptive name for a Passed Example.

			</a>
		</svg>
	</p>
</html>
```

### Failed

#### Failed Example 1

Both links have the same [accessible name][] and [context][programmatically determined link context] but go to different resources.

```html
<html lang="en">
	<p>
		<a href="http://facebook.com">Follow us</a>
		<a href="http://twitter.com">Follow us</a>
	</p>
</html>
```

#### Failed Example 2

The same [accessible name][] is used for two links in the same [context][programmatically determined link context] going to web pages that are similar, but have different information in their content.

```html
<html lang="en">
	<p>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
			>Contact us</a
		>
	</p>
</html>
```

#### Failed Example 3

These two HTML `span` elements have an [explicit role][] of link, but lead to resources that offer different content.

```html
<html lang="en">
	<p>
		<span
			role="link"
			tabindex="0"
			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html'"
		>

> 			Link text

Reviewer 4: We should use descriptive names.

<html lang="en">
	<p>
		<a href="http://facebook.com"><img src="facebook.jpg" alt="Follow us"/></a>
		<a href="http://twitter.com"><img src="twitter.jpg" alt="Follow us"/></a>
	</p>
</html>
```

#### Failed Example 5

A set of two SVG `a` elements that have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.

```html
<html lang="en">
	<p>
		<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
			<a href="http://facebook.com" aria-label="Follow us">
				<circle cx="50" cy="40" r="35" />
			</a>

			<a href="http://twitter.com">
				<text x="50" y="90" text-anchor="middle">
					Follow us
				</text>
			</a>
		</svg>
	</p>
</html>
```

#### Failed Example 6

Both links resolve to [same resource][] after redirect, but the redirect is not instant.

```html
<html lang="en">
	<p>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/redirect1.html">Contact us</a>
	</p>
</html>
```

### Inapplicable

#### Inapplicable Example 1

> These `a` and `area` elements have no `href` attribute.

Reviewer 4: ```suggestion
These `a` and `area` elements do not have a role of `link`.
```

---
id: fd3a94
name: Links with identical accessible names and context serve equivalent purpose
rule_type: atomic
description: |
  This rule checks that links with identical accessible names and context resolve to the same or equivalent resources.
accessibility_requirements:
  wcag20:2.4.4: # Link Purpose (In Context) (A)
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS Styling
  - Language
acknowledgments:
  authors:

>     - Anne Thyme Nørregaard

Reviewer 5: I think we should put Anne as previous author. She has been out of the loop 6 months.

>     - Anne Thyme Nørregaard

Reviewer 6: It might be worth moving some authors to previous authors.

## Expectation

When followed, the links in each set of target elements resolve to the [same resource][] or to [equivalent resources](#equivalent-resource).

**Note**: Resolving the links includes potential redirects, if the redirects happen instantly.

## Assumptions

- This rule assumes that the purpose of the links with identical [accessible names][accessible name] and [context][programmatically determined link context] would not be ambiguous to users in general, which is the exception mentioned in [Success Criterion 2.4.4 Link Purpose (In Context)][sc244]. If the links are ambiguous to users in general, users of assistive technologies are not at a disadvantage when viewing the links, which makes it more of a general user experience concern than an accessibility issue.
- This rule assumes that, within the context of the test subject, the description provided by the [accessible name][] of a link can only accurately describe one resource (notably, homonyms alone are not used as link names). Thus, if two or more links have the same [accessible name][] but resolve to different resources, at least one of them does not describe its purpose.

> - This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.

Reviewer 5: Language is an input. We don't need to assume we can determine it, if we can't, input is missing and the rule can't be executed. This can be taken out I think.

Reviewer 4: The CG decision was to keep this assumption nonetheless:
https://www.w3.org/2020/01/09-act-r-minutes.html#item04

- This rule assumes that assistive technologies are exposing all links on the page in the same way no matter which [document tree](https://dom.spec.whatwg.org/#document-trees) they are in. If an assistive technology requires the user to "enter" an `iframe` or a [shadow tree][] before exposing its links, then it is possible for two links to have identical name and context but resolve to different resources without failing [Success Criterion 2.4.4 Link Purpose (In Context)][sc244] (if said links are in separate [documents][document] or [shadow trees][shadow tree])

## Accessibility Support

_There are no major accessibility support issues known for this rule._

## Background

- [Understanding Success Criterion 2.4.4: Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
- [HTML Specification - URL parsing](https://html.spec.whatwg.org/#resolving-urls)

> ## Test Cases

Reviewer 5: I think you've overdone it a little on the test cases. The test cases aren't there to check if someone knows how to implement the accessible name algorithm. That's not what this rule is / should be about. I think several of these examples can be removed without.

Reviewer 7: OK. I'm pruning the examples.

### Passed

#### Passed Example 1

> These` two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] and link to the [same resource][].

Reviewer 5: ```suggestion
These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] and link to the [same resource][].
```

```html
<html lang="en">
	<p>
		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html"
			>Contact us</a
		>) and get in touch (
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>)
	</p>
</html>
```

#### Passed Example 2

> These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and resolve to the [same resource][] after instant redirect.

Reviewer 5: ```suggestion
These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and resolve to the [same resource][] after an instant redirect.
```

		>)
	</p>
</html>
```

#### Passed Example 5

These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and link to URLs that differ due to trailing slashes, but resolve to the [same resource][] after redirects caused by user agent.

```html
<html lang="en">
	<p>
		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/">Contact us</a>) and
		get in touch (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66">Contact us</a>)
	</p>
</html>
```

#### Passed Example 6

These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and go to pages that fulfill the same purpose in relation to the link because they contain the same relevant information.

```html
<html lang="en">
	<p>
		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html">Call us</a
		>) and get in touch (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page2.html"
			>Call us</a
		>)
	</p>
</html>
```

#### Passed Example 7

> These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and go to pages that use different layouts but have the same advertised key content.

Reviewer 5: ```suggestion
These two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context], and go to pages that use different layouts but have the same purpose.
```

			</a>
		</svg>
	</p>
</html>
```

### Failed

#### Failed Example 1

The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] but go to different resources.

```html
<html lang="en">
	<p>
		We are on social media:
		<a href="http://facebook.com">Follow us</a>
		<a href="http://twitter.com">Follow us</a>
	</p>
</html>
```

#### Failed Example 2

The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context]. They link to web pages that are similar, but have different information in their content.

```html
<html lang="en">
	<p>
		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html"
			>Contact us</a
		>) and get in touch (
		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/admissions/contact.html"
			>Contact us</a
		>)
	</p>
</html>
```

#### Failed Example 3

The two HTML `span` elements have an [explicit role][] of link, same [accessible name][] and [context][programmatically determined link context], but lead to resources that offer different content.

> 

Reviewer 3: this example seems to me to be "ambiguous to all users"

Reviewer 7: Agree that each link can be ambiguous to all users. Updated the text to be similar to what is used in some of the other examples.

	</p>
</html>
```

#### Failed Example 4

The two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.

```html
<html lang="en">
	<p>
		We are on social media:
		<a href="http://facebook.com"><img src="facebook.jpg" alt="Follow us"/></a>
		<a href="http://twitter.com"><img src="twitter.jpg" alt="Follow us"/></a>
	</p>
</html>
```

#### Failed Example 5

The two SVG `a` elements have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.

```html
<html lang="en">
	<p>
		<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
			<a href="http://facebook.com" aria-label="Follow us">
				<circle cx="50" cy="40" r="35" />
			</a>

			<a href="http://twitter.com">
				<text x="50" y="90" text-anchor="middle">
					Follow us
				</text>
			</a>
		</svg>
	</p>
</html>
```

#### Failed Example 6

The two HTML `a` elements with the same [accessible name][] and [context][programmatically determined link context] resolve to the [same resource][] after redirect, but the redirect is not instant.

> 

Reviewer 3: isn't what counts the end result? not sure i agree with this as a failure

Reviewer 7: But we can't tell when the redirect will happen if it does not happen instantly. That's why we have a note stating "Resolving the links includes potential redirects, if the redirects happen instantly."

Imagine a redirect that happens half an hour after the link has been followed. Almost all users will not even perceive that redirect, so they will experience different content and the SC should fail.

(of course we can tell when the redirect happens, but then the issue becomes where do we draw the line between redirects that will be perceived and those that won't)

rule_type: atomic
description: |
  This rule checks that links with identical accessible names and context resolve to the same resource or equivalent resources.
accessibility_requirements:
  wcag20:2.4.4: # Link Purpose (In Context) (A)
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed
input_aspects:
  - DOM Tree
  - CSS Styling
  - Language
acknowledgements:
  authors:
    - Anne Thyme Nørregaard
    - Carlos Duarte
---

## Applicability

This rule applies to any set of any two or more HTML or SVG elements which

- have the [semantic role][] of `link`, or a role that inherits from the `link` role; and
- are in the same [web page (HTML)][]; and
- are [included in an accessibility tree][included in the accessibility tree]; and
- have [matching][] [accessible names][accessible name] that are not empty (`""`); and
- have the same [programmatically determined link context][].

**Note:** The test target for this rule is the full set of link elements that share the same [matching][] [accessible name][] and [programmatically determined link context][].

## Expectation

> When followed, the links in each set of target elements resolve to the [same resource][] or to [equivalent resources](#equivalent-resource). Resolving the links includes potential redirects, if the redirects happen instantly.

Reviewer 5: I think this is better. That second sentence does not add to the expectation, it just clarifies it.

```suggestion
When followed, the links in each set of target elements resolve to the [same resource][] or to [equivalent resources](#equivalent-resource).

**Note**: Resolving the links includes potential redirects, if the redirects happen instantly.
```

- This rule assumes that, within the context of the test subject, the description provided by the [accessible name][] of a link can only accurately describe one resource (notably, homonyms alone are not used as link names). Thus, if two or more links have the same [accessible name][] but resolve to different resources, at least one of them does not describe its purpose.
- This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.

## Accessibility Support

This rule assumes that assistive technologies are exposing all links on the page in the same way no matter which [document tree](https://dom.spec.whatwg.org/#document-trees) they are in. If an assistive technology requires the user to "enter" an `iframe` or a [shadow tree][] before exposing its links, then it is possible for two links to have identical name and context but resolve to different resources without failing [Success Criterion 2.4.4 Link Purpose (In Context)][sc244] (if said links are in separate [documents][document] or [shadow trees][shadow tree])

## Background

> - [CSS Scoping Module Level 1 (editor's draft)](https://drafts.csswg.org/css-scoping/)

Reviewer 5: How does this related to it? Can we link to the relevant section if this is necessary?

Reviewer 7: Not needed.

- [Understanding Success Criterion 2.4.4: Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
- [HTML Specification - URL parsing](https://html.spec.whatwg.org/#resolving-urls)

## Test Cases

### Passed

#### Passed Example 1

> A set of two HTML `a` elements have the same [accessible name][] and [context][programmatically determined link context] and link to the [same resource][].

Reviewer 5: Can you make sure these example descriptions follow the format outlined here: https://github.com/act-rules/act-rules.github.io/pull/946/files#diff-9772ff397ce17764837b9ae6f1fd71acR142

Reviewer 7: Done

	</p>
</html>
```

#### Failed Example 5

A set of two SVG `a` elements that have the same [accessible name][] and [context][programmatically determined link context] but link to different resources.

```html
<html lang="en">
	<p>
		<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
			<a href="http://facebook.com" aria-label="Follow us">
				<circle cx="50" cy="40" r="35" />
			</a>

			<a href="http://twitter.com">
				<text x="50" y="90" text-anchor="middle">
					Follow us
				</text>
			</a>
		</svg>
	</p>
</html>
```

#### Failed Example 6

Both links resolve to [same resource][] after redirect, but the redirect is not instant.

```html
<html lang="en">
	<p>
		Learn more (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html"
			>Contact us</a
		>) and get in touch (<a
			href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/redirect1.html"
			>Contact us</a
		>)
	</p>
</html>
```

### Inapplicable

#### Inapplicable Example 1

These `a` and `area` elements do not have a role of `link`.

```html
<html lang="en">
	<a>Link text</a>

> 	<area aria-label="Link text" />

Reviewer 5: That's two test cases in one. Can you split them out please?

Reviewer 7: Not sure I understand what you mean. What is being tested here besides elements that do not have the link role?

	</p>
</html>
```

#### Inapplicable Example 3

These links have the same [accessible name][] and link to the [same resource][] but different [programmatically determined link contexts][programmatically determined link context].

```html
<html lang="en">
	<ul>
		<li>
			To learn more about us:
			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>
		</li>
		<li>
			To get in touch with us:
			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>
		</li>
	</ul>
</html>
```

#### Inapplicable Example 4

These `span` elements do not have a [semantic role][] of link.

```html
<html lang="en">
	<p>
		Learn more (<span
			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html'"
		>
			Contact Us </span
		>) and get in touch (<span
			onclick="location='/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page2.html'"
		>
			Contact Us </span
		>)
	</p>
</html>
```

#### Inapplicable Example 5

These links have empty [accessible names][accessible name].

```html

> <a href="http://facebook.com"></a> <a href="http://twitter.com"></a>

Reviewer 5: These are inaccessible links. I don't think these satisfy the SC. We should make sure all inapplicable examples satisfy the SC.

Reviewer 7: Links now satisfy the SC.

```

#### Inapplicable Example 6

These image links have empty [accessible names][accessible name].

```html

> <a href="http://facebook.com"><img src="facebook.jpg"/></a> <a href="http://twitter.com"><img src="twitter.jpg"/></a>

Reviewer 5: As above, I don't think these satisfy the SC.

Reviewer 7: Removed this example, given that it was testing the same as the previous.

		<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a>
	</p>
</html>
```

#### Inapplicable Example 3

These links have the same [accessible name][] and link to the [same resource][] but different [programmatically determined link contexts][programmatically determined link context].

```html
<html lang="en">
	<ul>
		<li>
			To learn more about us:
			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>
		</li>
		<li>
			To get in touch with us:
			<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/index.html">Contact us</a>
		</li>
	</ul>
</html>
```

#### Inapplicable Example 4

The first link is not [included in the accessibility tree][].

```html
<html lang="en">
	<p>
		Learn more (<a
			href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page1.html"
			aria-hidden="true"
			tabindex="-1"
			>Contact Us</a
		>) and get in touch (<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/page2.html"
			>Contact Us</a
		>)
	</p>
</html>
```

> 

Reviewer 4: Note: the first link is actually included in the Accessibility Tree (with a role of `link`) in Chrome who chooses to expose focusable elements over `aria-hidden` (Edge and Firefox will prioritise `aria-hidden`). This is a part of the spec that is actually not specified but is linked to presentational role conflict resolution.

I am working on a PR to try and fix this problem for us. In the meantime, I suggest removing examples with `aria-hidden` on focusable elements (`tabindex=-1` removes it from sequential order, but it is still focusable). Because the behaviour is not specified and browsers disagree (so we can neither follow the spec nor the de facto industry standard).

See #1189 for my work in progress on that topic and links to more discussion…

Reviewer 7: Removed it

#### Inapplicable Example 5

The two HTML `a` elements have empty [accessible names][accessible name].

```html

> <a href="http://facebook.com"></a> <a href="http://facebook.com"><img src="facebook.jpg"/></a>

Reviewer 5: Neither of these are accessible links. Can you make sure this satisfies the SC?

Reviewer 7: Modified the example so that they are both in the accessibility tree but only one has an accessible name.
